packages/connection-form/src/utils/csfle-kms-fields.ts

```diff
@@ -0,0 +1,172 @@
+    optional: false,
+    value: (autoEncryption) => autoEncryption?.kmsProviders?.gcp?.email ?? '',
+    state: 'none',
+    description: 'The service account email to authenticate.',
+  },
+  {
+    name: 'privateKey',
+    label: 'Private Key',
+    type: 'textarea',
+    optional: false,
+    value: (autoEncryption) =>
+      autoEncryption?.kmsProviders?.gcp?.privateKey?.toString('base64') ?? '',
+    state: 'none',
+    description: 'A base64-encoded PKCS#8 private key.',
+  },
+  {
+    name: 'endpoint',
+    label: 'Endpoint',
+    type: 'text',
+    optional: true,
+    value: (autoEncryption) =>
+      autoEncryption?.kmsProviders?.gcp?.endpoint ?? '',
+    state: 'none',
+    description: 'A host with optional port.',
+  },
+];
+
+const AWSFields: KMSField<'aws'>[] = [
+  {
+    name: 'accessKeyId',
+    label: 'Access Key ID',
+    type: 'text',
+    optional: false,
+    value: (autoEncryption) =>
+      autoEncryption?.kmsProviders?.aws?.accessKeyId ?? '',
+    state: 'none',
+    description: 'The access key used for the AWS KMS provider.',
+  },
+  {
+    name: 'secretAccessKey',
+    label: 'Secret Access Key',
+    type: 'password',
+    optional: false,
+    value: (autoEncryption) =>
+      autoEncryption?.kmsProviders?.aws?.secretAccessKey ?? '',
+    state: 'none',
+    description: 'The secret access key used for the AWS KMS provider.',
+  },
+  {
+    name: 'sessionToken',
+    label: 'Session Token',
+    type: 'password',
+    optional: true,
+    value: (autoEncryption) =>
+      autoEncryption?.kmsProviders?.aws?.sessionToken ?? '',
+    state: 'none',
+    description:
+      'An optional AWS session token that will be used as the X-Amz-Security-Token header for AWS requests.',
+  },
+];
+
+const AzureFields: KMSField<'azure'>[] = [
+  {
+    name: 'tenantId',
+    label: 'Tenant ID',
+    type: 'text',
+    optional: false,
+    value: (autoEncryption) =>
+      autoEncryption?.kmsProviders?.azure?.tenantId ?? '',
+    state: 'none',
+    description: 'The tenant ID identifies the organization for the account.',
+  },
+  {
+    name: 'clientId',
+    label: 'Client ID',
+    type: 'text',
+    optional: false,
+    value: (autoEncryption) =>
+      autoEncryption?.kmsProviders?.azure?.clientId ?? '',
+    state: 'none',
+    description: 'The client ID to authenticate a registered application.',
+  },
+  {
+    name: 'clientSecret',
+    label: 'Client Secret',
+    type: 'password',
+    optional: false,
+    value: (autoEncryption) =>
+      autoEncryption?.kmsProviders?.azure?.clientSecret ?? '',
+    state: 'none',
+    description: 'The client secret to authenticate a registered application.',
+  },
+  {
+    name: 'identityPlatformEndpoint',
+    label: 'Identity Platform Endpoint',
+    type: 'text',
+    optional: true,
+    value: (autoEncryption) =>
+      autoEncryption?.kmsProviders?.azure?.identityPlatformEndpoint ?? '',
+    state: 'none',
+    description: 'A host with optional port.',
```

Comment:
Same as above, maybe:
```suggestion
    description: 'A host with an optional port.',
```
reads better? or maybe:
```suggestion
    description: 'A host with port optional.',
```